stock_inventory/models/stock_inventory.py

```diff
@@ -191,24 +191,55 @@ def _get_domain_category_quants(self, base_domain):
             ]
         )
 
+    def _get_products(self):
+        self.ensure_one()
+        if self.product_selection == "all":
+            return self.env["product.product"].search([])
+        elif self.product_selection == "category" and self.category_id:
+            return self.env["product.product"].search(
+                [("categ_id", "=", self.category_id.id)]
+            )
+        else:
+            return self.product_ids
+
+    def _get_overlapping_inventories(self):
+        self.ensure_one()
```

Comment:
I'm just wondering why not using multi. As this function is very interesting. Maybe to build a report, or...

Reply:
Well, it is not needed for the purpose of the change, but yes, it may be useful for other purposes, so ok to me.